src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializersTests.cs

```diff
@@ -137,5 +138,391 @@ namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : S
 C.M
 Program.Main");
         }
+
+        [Fact]
+        public void MultipleInitializers_SingleFile()
+        {
+            string source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class C1
+{
+    [ModuleInitializer]
+    internal static void M1() => Console.Write(1);
+
+    internal class C2
+    {
+        [ModuleInitializer]
+        internal static void M2() => Console.Write(2);
+    }
+
+    [ModuleInitializer]
+    internal static void M3() => Console.Write(3);
+}
+
+class Program 
+{
+    static void Main() => Console.Write(4);
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+
+            CompileAndVerify(
+                source,
+                parseOptions: s_parseOptions,
+                options: TestOptions.DebugExe.WithMetadataImportOptions(MetadataImportOptions.All),
+                expectedOutput: "1234");
+        }
+
+        [Fact]
+        public void MultipleInitializers_DifferentContainingTypeKinds()
+        {
+            string source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class C1
+{
+    [ModuleInitializer]
+    internal static void M1() => Console.Write(1);
+}
+
+struct S1
+{
+    [ModuleInitializer]
+    internal static void M2() => Console.Write(2);
+}
+
+interface I1
+{
+    [ModuleInitializer]
+    internal static void M3() => Console.Write(3);
+}
+
+class Program 
+{
+    static void Main() => Console.Write(4);
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+
+            CompileAndVerify(
+                source,
+                parseOptions: s_parseOptions,
+                targetFramework: TargetFramework.NetStandardLatest,
+                options: TestOptions.DebugExe.WithMetadataImportOptions(MetadataImportOptions.All),
```

Comment:
I don't believe these lines serve a purpose unless a `symbolValidator` is provided.